apollo/src/graphql/model-version/modelVersionMutations.ts

```diff
@@ -151,13 +176,27 @@ builder.mutationFields((t) => ({
                 // Intentionally don't throw an error here on archived storage
                 // providers or models. It's not unreasonable to want to mark old
                 // assets as archived if a parent object goes bye-bye
-                const mlModelVersion = await ObjectionMLModelVersion.query(trx).updateAndFetchById(
-                    args.modelVersionId,
-                    {
-                        isArchived: raw('NOT is_archived'),
-                    },
+                const mlModelVersion = await ObjectionMLModelVersion.query().findById(args.modelVersionId);
+                if (mlModelVersion === undefined) {
+                    throw new RegistryOperationError({ name: 'VERSION_PERMISSION_ERROR' });
+                }
+
+                const parentModel = (await ObjectionMLModel.query()
+                    .where('modelId', mlModelVersion.modelId)
+                    .first()) as ObjectionMLModel;
+                const team = await ObjectionMLModel.relatedQuery('getTeam')
+                    .for(parentModel.modelId)
+                    .first() as ObjectionTeam;
+                await ObjectionTeamEdge.userHasRole(
+                    ctx.user.$id(),
+                    team.$id(),
+                    ['owner', 'member']
                 );
 
+                await mlModelVersion.$query(trx).patchAndFetch({
```

Comment:
to be fair I need to test and make sure this actually returns the updated result set properly. I'm just being lazy for now 😬

Reply:
The best way to be!